pkg/controllers/inventory/inventory_controller.go

```diff
@@ -0,0 +1,216 @@
+package inventory
+
+import (
+	"context"
+	"errors"
+	"sync"
+	"time"
+
+	v1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/util/wait"
+	"k8s.io/client-go/tools/cache"
+	"k8s.io/client-go/util/workqueue"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	"github.com/vmware-tanzu/nsx-operator/pkg/config"
+	"github.com/vmware-tanzu/nsx-operator/pkg/logger"
+	commonservice "github.com/vmware-tanzu/nsx-operator/pkg/nsx/services/common"
+	"github.com/vmware-tanzu/nsx-operator/pkg/nsx/services/inventory"
+)
+
+const (
+
+	// How long to wait before retrying
+	minRetryDelay = 5 * time.Second
+	maxRetryDelay = 300 * time.Second
+
+	inventoryGCJitterFactor = 0.1
+)
+
+var (
+	log     = &logger.Log
+	keyFunc = cache.DeletionHandlingMetaNamespaceKeyFunc
+)
+
+type InventoryController struct {
+	Client               client.Client
+	service              *inventory.InventoryService
+	inventoryObjectQueue workqueue.TypedRateLimitingInterface[any]
+	keyBuffer            inventory.KeySet
+	inventoryMutex       sync.Mutex
+	cf                   *config.NSXOperatorConfig
+}
+
+func NewInventoryController(Client client.Client, service *inventory.InventoryService, cf *config.NSXOperatorConfig) *InventoryController {
+	c := &InventoryController{
+		Client:               Client,
+		service:              service,
+		keyBuffer:            inventory.KeySet{},
+		cf:                   cf,
+		inventoryObjectQueue: workqueue.NewTypedRateLimitingQueueWithConfig(workqueue.NewTypedItemExponentialFailureRateLimiter[any](minRetryDelay, maxRetryDelay), workqueue.TypedRateLimitingQueueConfig[any]{Name: "inventoryObject"})}
+	return c
+}
+
+// Initialize sync NSX resources and update local inventory store.
+// TODO: Add two E2E tests to cover adding a new cluster and updating an existing container cluster.
+func (c *InventoryController) Initialize() error {
+	return nil
+}
+
+func (c *InventoryController) handlePod(obj interface{}) {
+	var pod *v1.Pod
+	ok := false
+	switch obj1 := obj.(type) {
+	case *v1.Pod:
+		pod = obj1
+	case cache.DeletedFinalStateUnknown:
+		pod, ok = obj1.Obj.(*v1.Pod)
+		if !ok {
+			log.Error(errors.New("Unknown obj"), "DeletedFinalStateUnknown Obj is not *v1.Pod")
+			return
+		}
+	}
+	log.V(1).Info("Inventory processing Pod", "namespace", pod.Namespace, "name", pod.Name)
+	key, _ := keyFunc(pod)
+	log.V(1).Info("Adding Pod key to inventory object queue", "pod key", key)
```

Comment:
Could you also update other occurrence of pod to Pod?
```suggestion
	log.V(1).Info("Adding Pod key to inventory object queue", "Pod key", key)
```

Reply:
Done